src/presign/round_one.rs

```diff
@@ -22,16 +22,39 @@ use serde::{Deserialize, Serialize};
 use std::fmt::Debug;
 use zeroize::ZeroizeOnDrop;
 
+use super::round_three::SecretBigNumber;
+
+#[derive(Clone, ZeroizeOnDrop)]
+pub(crate) struct SecretNonce(Nonce);
+
+impl SecretNonce {
+    pub fn get_nonce(&self) -> &Nonce {
+        &self.0
+    }
+}
+
+impl From<Nonce> for SecretNonce {
+    fn from(nonce: Nonce) -> Self {
+        SecretNonce(nonce)
+    }
+}
+
+/*impl From<SecretNonce> for Nonce {
+    fn from(secret_nonce: SecretNonce) -> Self {
+        secret_nonce.clone().get_nonce()
+    }
+}*/
+
 /// Private data used in round one of the presign protocol.
-#[derive(ZeroizeOnDrop)]
+//#[derive(ZeroizeOnDrop)]
 pub(crate) struct Private {
-    pub k: BigNumber,
-    pub rho: Nonce,
+    pub k: SecretBigNumber,
+    pub rho: SecretNonce,
     pub gamma: BigNumber,
-    pub nu: Nonce,
-    #[zeroize(skip)]
+    pub nu: SecretNonce,
+    //#[zeroize(skip)]
```

Comment:
Any reason you commented these out?

Reply:
I commented these out because the ZeroizeOnDrop was removed entirely from the whole Private structure. So was thinking if skipping zeroize makes sense here after that change?